### PR TITLE
Added ingress service type variable (ClusterIP or Loadbalancer)

### PR DESCRIPTION
**Pull request by Barteus**
This is PR based on: https://github.com/canonical/istio-operators/pull/67

This PR allows exposing applications behind the Istio gateway via  Ingress Controller & rule

### CODE REVIEW

File: charms/istio-gateway/src/manifest.yaml [hunk 1]
```diff
@@ -318,4 +318,4 @@ spec:
       targetPort: 8443
   selector:
     istio: {{ kind }}gateway
-  type: {{ 'ClusterIP' if kind == 'egress' else 'LoadBalancer' }}
+  type: {{ 'ClusterIP' if kind == 'egress' else ingress_service_type }}
```
Comment (DomFleischmann) [inline charms/istio-gateway/src/manifest.yaml:321]: Lets remove the egress part here also.

**Review by DomFleischmann — COMMENTED** [review]
Looks promising, just some minor comments.

File: charms/istio-gateway/config.yaml [hunk 2]
```diff
@@ -3,3 +3,8 @@ options:
     default: ''
     description: Kind of Istio gateway. Must be set to one of `ingress`, `egress`
     type: string
+  ingress_service_type:
+    default: 'LoadBalancer'
+    description: |
+      Type of service for the ingress gateway out of: 'ClusterIP' or 'LoadBalancer'.
```
[the remark this answers is not in the document]
Comment (Barteus) [reply charms/istio-gateway/config.yaml]: How do you propose to do this, please share.

File: charms/istio-gateway/src/manifest.yaml (hunk 1 above)
Comment (Barteus) [reply charms/istio-gateway/src/manifest.yaml:321]: The rule of the good scout says to leave in the better share than found, but I would like to separate removing "kind" from this charm and adding a new configuration option.

File: charms/istio-gateway/config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Barteus) [reply charms/istio-gateway/config.yaml]: we can use the lowercase value in the config and remap it in the Charm code to LoadBalancer or ClusterIP.

The downside of it is that if you see "LoadBalancer" or "ClusterIP" you know out of the box that its the Kubernetes service type.

Its a trade-off.

**Pull request merged**